include/fiction/technology/charge_distribution_surface.hpp

```diff
@@ -828,10 +1496,455 @@ class charge_distribution_surface<Lyt, false> : public Lyt
 
             for (uint64_t i = 0u; i < strg->pot_mat.size(); ++i)
             {
-                strg->loc_pot[i] += -(this->get_chargless_potential_by_indices(i, random_element));
+                strg->local_pot[i] += -(this->get_chargless_potential_by_indices(i, random_element));
             }
         }
     }
+    /**
+     * This function can be used to assign a global external electrostatic potential in Volt (unit: V) to the layout
+     * (e.g this could be a planar external electrode).
+     *
+     * @param potential_value Value of the global external electrostatic potential in Volt (e.g. -0.3).
+     * Charge-transition levels are shifted by this value.
+     */
+    void assign_global_external_potential(const double        potential_value,
+                                          dependent_cell_mode dependent_cell = dependent_cell_mode::FIXED) noexcept
+    {
+        this->foreach_cell(
+            [this, &potential_value](const auto& cell) {
+                strg->local_external_pot.insert({cell, potential_value});
+            });
+        this->update_after_charge_change(dependent_cell);
+    }
+    /**
+     * This function determines if given layout has to be simulated with three states since positively charged SiDBs can
+     * occur due to the local potential analysis.
+     *
+     * @return return value is true when three state simulation is required.
+     */
+    bool three_state_simulation_required() noexcept
```

Comment:
Based on the function name, it is unclear what this function does, because it is lacking a verb. Maybe add `is_` as a prefix?